honeycombio/data_source_query_specification.go

```diff
@@ -353,49 +353,55 @@ func extractFilter(d *schema.ResourceData, index int) (honeycombio.FilterSpec, e
 	filter.Op = honeycombio.FilterOp(d.Get(fmt.Sprintf("filter.%d.op", index)).(string))
 
 	valueSet := false
-	v, vOk := d.GetOk(fmt.Sprintf("filter.%d.value", index))
-	if vOk {
-		filter.Value = coerceValueToType(v.(string))
+	if v, ok := d.GetOk(fmt.Sprintf("filter.%d.value", index)); ok {
+		if filter.Op == honeycombio.FilterOpIn || filter.Op == honeycombio.FilterOpNotIn {
+			// if the filter operation is 'in' or 'not-in', just leave the value as a string
+			filter.Value = v
+		} else {
+			filter.Value = coerceValueToType(v.(string))
+		}
 		valueSet = true
 	}
-	vs, vsOk := d.GetOk(fmt.Sprintf("filter.%d.value_string", index))
-	if vsOk {
+	if v, ok := d.GetOk(fmt.Sprintf("filter.%d.value_string", index)); ok {
 		if valueSet {
 			return filter, fmt.Errorf(multipleValuesError)
 		}
-		filter.Value = vs
+		filter.Value = v
 		valueSet = true
 	}
-	vi, viOk := d.GetOk(fmt.Sprintf("filter.%d.value_integer", index))
-	if viOk {
+	if v, ok := d.GetOk(fmt.Sprintf("filter.%d.value_integer", index)); ok {
 		if valueSet {
 			return filter, fmt.Errorf(multipleValuesError)
 		}
-		filter.Value = vi
+		filter.Value = v
 		valueSet = true
 	}
-	vf, vfOk := d.GetOk(fmt.Sprintf("filter.%d.value_float", index))
-	if vfOk {
+	if v, ok := d.GetOk(fmt.Sprintf("filter.%d.value_float", index)); ok {
 		if valueSet {
 			return filter, fmt.Errorf(multipleValuesError)
 		}
-		filter.Value = vf
+		filter.Value = v
 		valueSet = true
 	}
-	vb, vbOk := d.GetOk(fmt.Sprintf("filter.%d.value_boolean", index))
-	if vbOk {
+	if v, ok := d.GetOk(fmt.Sprintf("filter.%d.value_boolean", index)); ok {
 		if valueSet {
 			return filter, fmt.Errorf(multipleValuesError)
 		}
-		filter.Value = vb
+		filter.Value = v
 	}
 
 	if filter.Op == honeycombio.FilterOpIn || filter.Op == honeycombio.FilterOpNotIn {
```

Comment:
I understand we're doing "it's an in,not-in" filter things in two places at the moment, but this second check is a function of the deprecated `value_*` options. Once they're removed we can do this all in one place above